docs/references/go/verifying-sessions.mdx

```diff
@@ -1,102 +1,161 @@
 ---
-title: Veryifing a session | Clerk Go SDK
+title: Verify a Clerk session in Go
 description: Learn how to verify a session with Clerk in your Go application.
 ---
 
-# Verifying a session in Go
+# Verify a Clerk session in Go
 
-## Protect your backend APIs
+There are two ways to verify a session with Clerk in your Go application:
+- [Using Clerk middleware](#use-clerk-middleware-to-verify-a-session)
+If you want to verify a session in an HTTP context, it is recommended to use Clerk middleware. Clerk middleware guarantees better performance and efficiency by making the minimum necessary requests to the Clerk Backend API.
 
-Go makes it really easy to create a simple HTTP server, and Clerk makes it simple to authenticate any request. The following examples shows how to verify a session and retrieve the corresponding user.
+- [Manually verifying the session token](#manually-verify-a-session-token)
+If you want to verify a session in a non-HTTP context, or if you would like total control over the verification process, you can verify the session token on your own.
 
-<InjectKeys>
+## Use Clerk middleware to verify a session
 
-```go
-package main
+Go enables you to create a simple HTTP server, and Clerk enables you to authenticate any request. Together, you can create a secure server that only allows authenticated users to access certain routes.
+
+Clerk Go SDK provides two functions for adding authentication to HTTP handlers:
+
+- [`WithHeaderAuthorization()`](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#WithHeaderAuthorization)
+- [`RequireHeaderAuthorization()`](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2/http#RequireHeaderAuthorization), which calls `WithHeaderAuthorization()` under the hood, but responds with `HTTP 403 Forbidden` if it fails to detect valid session claims.
+
+Both middleware functions support header based authentication with a bearer token. The token will be parsed, verified, and its claims will be extracted as [`SessionClaims`](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaims).
+
+The claims will then be made available in the `http.Request.Context` for the next handler in the chain. Clerk Go SDK provides the [`SessionClaimsFromContext()`](https://pkg.go.dev/github.com/clerk/clerk-sdk-go/v2#SessionClaimsFromContext) helper for accessing the claims from the context.
+
+The following example demonstrates how to use the `WithHeaderAuthorization()` middleware to protect a route. If the user tries accessing the route and their session token is valid, the user's ID will be returned in the response.
 
+<Callout type="info">
+Your Clerk secret key is required. If you are signed in to your Clerk Dashboard, your secret key should become visible by clicking on the eye icon. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.
```

Comment:
```suggestion
Your Clerk secret key is required. If you are signed in to your Clerk Dashboard, your secret key should become visible by selecting the eye icon. Otherwise, you can retrieve your Clerk secret key from the Clerk Dashboard on the **[API Keys](https://dashboard.clerk.com/last-active?path=api-keys)** page.
```